database-seed/mongo-loader.js

```diff
@@ -9,43 +9,43 @@ mongoose.connect(DATABASE_LINK, { useNewUrlParser: true, useCreateIndex: true })
 let Product = mongoose.model('ProductImage', DB_SCHEMA);
 
 const seedDatabase = () => {
-    dropDatabase()
+  dropDatabase()
     .then(() => {
-        let documents = JSON.parse(retrieveData());
-        let updates = [];
-        documents.forEach((document) => {
-            updates.push(update(document));
-        });
-        Promise.all(updates).then(() => {
-            console.log('Database seeded');
-            mongoose.connection.close();
-
-        })
+      let documents = JSON.parse(retrieveData());
+      let updates = [];
+      documents.forEach((document) => {
+        updates.push(update(document));
+      });
+      Promise.all(updates).then(() => {
+        console.log('Database seeded');
+        mongoose.connection.close();
+
+      })
     });
-}
+};
 
 const update = (document) => {
-    return new Promise((resolve, reject) => {
-        Product.updateOne({'productID':document.productID}, document, {upsert: true}, (err) => {
-            if (err) reject(err);
-            resolve();
-        });
+  return new Promise((resolve, reject) => {
+    Product.updateOne({'productID': document.productID}, document, {upsert: true}, (err) => {
```

Comment:
what is `upsert` ? and also what is this `update` function needed for?

Reply:
upsert allows you to, in one motion:
1) check if a matching document exists
2) if it doesnt, insert the new document
3) if it does exist, then update any changes to the document.

Otherwise, with mongo, you would need to do:
1. `Query` for a document
2. if no result, then call `Save`
3. if not empty, call `update`

Reply:
`updateOne` is the new method in mongo.  It used to be called `update`, but they've split it to `updateOne` and `updateMany`